test/units/galaxy/test_collection.py

```diff
@@ -350,43 +342,75 @@ def test_validate_certs_with_server_url(global_ignore_certs, monkeypatch):
         '-s',
         'https://galaxy.ansible.com'
     ]
-    if global_ignore_certs:
+    if ignore_certs_cli:
         cli_args.append('--ignore-certs')
+    if ignore_certs_cfg is not None:
+        monkeypatch.setattr(C, 'GALAXY_IGNORE_CERTS', ignore_certs_cfg)
 
     galaxy_cli = GalaxyCLI(args=cli_args)
     mock_execute_install = MagicMock()
     monkeypatch.setattr(galaxy_cli, '_execute_install_collection', mock_execute_install)
     galaxy_cli.run()
 
     assert len(galaxy_cli.api_servers) == 1
-    assert galaxy_cli.api_servers[0].validate_certs is not global_ignore_certs
+    validate_certs = (False if ignore_certs_cli else not ignore_certs_cfg if ignore_certs_cfg is not None else True)
+    assert galaxy_cli.api_servers[0].validate_certs == validate_certs
 
 
-@pytest.mark.parametrize('global_ignore_certs', [True, False])
-def test_validate_certs_with_server_config(global_ignore_certs, server_config, monkeypatch):
-
-    # test sidesteps real resolution and forces the server config to override the cli option
-    get_plugin_options = MagicMock(side_effect=server_config)
-    monkeypatch.setattr(C.config, 'get_plugin_options', get_plugin_options)
-
+@pytest.mark.parametrize(
+    # False/True/omitted, True/omitted
+    "ignore_certs_cli,ignore_certs_cfg",
+    [
+        (None, None),
+        (None, True),
+        (None, False),
+        (True, None),
+        (True, True),
+        (True, False),
+    ]
+)
+def test_validate_certs_server_config(ignore_certs_cfg, ignore_certs_cli, monkeypatch):
```

Comment:
Having stared at this long enough, my suggestion is to put the expectation into params like this, instead of having complex conditional assertions at the end of the test function:
```suggestion
    ("ignore_certs_cli", "ignore_certs_cfg", "expected_api_server2_validate_certs"),
    (
        (None, None, True),
        (None, True, False),
        (None, False, True),
        (True, None, False),
        (True, True, False),
        (True, False, False),
    ),
)
def test_validate_certs_server_config(ignore_certs_cfg, ignore_certs_cli, expected_api_server2_validate_certs, monkeypatch):
```

To produce a nice test run report, this could also make use of `pytest.param(..., id=)` or `ids=`.